src/app/teams/teams.service.ts

```diff
@@ -27,16 +27,54 @@ export class TeamsService {
     }));
   }
 
-  addToTeam(uid: string) {
-    return this.membersCollection.doc<Membership>(uid).update({
+  addToTeam(membership: Membership) {
```

Comment:
you don't have to pass the whole Membership object here, only the membership id as string is enough
```suggestion
  addToTeam(id: string) {
```
then you can get rid of 
`const id = membership.uid + membership.teamId;`